WormBase/names
# Bugfix/spec validation for gene update

Hi @sibyl229,

The main change here is to the `spec` used for update gene (now same as the one for new) - which enforces that a gene cannot be edit to something other than a "cloned" or "uncloned" representation.

wormbase.names.coercion: why it's needed:
 - By default, compojure-api's spec implementation strips undefined keys from a spec.
   This would be desirable in most scenarios, but in our case where we have all attributes and two permissable sub-sets, we don't want that.  ATM it's a bit of a chunk of code to change this behaviour, will look into putting a PR for that in upstream.  

The majority of the tests were using the old (incorrect) spec for update for generating input data,
hence all the changes to the tests.

- Steps for testing

   - testing: cloned gene 
     - try blanking/removing sequence name and click edit, should result in validation error

  - testing: uncloned gene
    - try adding a biotype and submitting 

Fixes #75 and provides does the "right thing" with respect to #72
